### PR TITLE
FCCEUD-33: Describe roles

### PR DESCRIPTION
**Pull request by theashiot**
* Create a concept module User roles in Pantheon

### CODE REVIEW

**Comment by theashiot** [comment]
@redhataccess/pantheon-developers Please review for technical completeness and accuracy.

**Comment by benradey** [comment]
Thumbs up from me!

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[id="user-roles-in-pantheon_{context}"]
+= User roles in {ProductShortName}
+
+In {ProductShortName}, each user is assigned a role that determines the actions the user can perform.
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:4]: In general, use 2nd person. Suggest:
`Each {ProductShortName} user is assigned a role that determines what actions you can perform.`

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+There are two user roles defined in {ProductShortName} - `Author` and `Publisher`. The following permissions are associated with the roles:
+
+`Author`::
+
+* Use Git to import their modules files into {ProductShortName}.
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:10]: Suggestion.
* `Import content`

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+* Revise existing modules.
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:11]: Replace `modules` with `content` everywhere, unless the action really does apply just to modules.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/attributes.adoc [hunk 4]
```diff
@@ -11,5 +11,6 @@
 
 // Publishing
 :PublishingPortal: Customer Portal
+:PublishingPortalSearh: Customer Portal Search
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/attributes.adoc:14]: Did you mean to include a `c` PublishingPortalSear**c**h?

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+* Preview and stage modules.
+* Search and sort any modules imported into {ProductShortName}.
+* Manage module metadata that is configurable in the user interface.
+
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:15]: Authors can also create product versions (as opposed to products).

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+* Revise existing modules.
+* Preview and stage modules.
+* Search and sort any modules imported into {ProductShortName}.
+* Manage module metadata that is configurable in the user interface.
+ 
+`Publisher`::
+* All the permissions associated with the `Author` role.
+* Publish modules to the {PublishingPortal} and the {PublishingPortalSearh}. 
+* Unpublish modules from the {PublishingPortal} and the {PublishingPortalSearh}.
+* Create and manage `Product` and `Product Version`.
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:20]: Authors can also create product versions. (At least I was able to on staging when signed in with only author privileges.)

I wouldn't format **Product** or **Product Version** as code. Just stay with unformatted, lower case product and product version.

**Review by stoobie — CHANGES_REQUESTED** [review]

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 7]
```diff
@@ -0,0 +1,20 @@
+[id="user-roles-in-pantheon_{context}"]
+= User roles in {ProductShortName}
+
+In {ProductShortName}, each user is assigned a role that determines the actions the user can perform.
+
+There are two user roles defined in {ProductShortName} - `Author` and `Publisher`. The following permissions are associated with the roles:
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:6]: I just checked RHV documentation, and I see that the roles there are in bold. In any case, code formatting is not the right markup in this context. I suggest marking up the roles like so: **Author** and **Publisher**.

**Review by stoobie — CHANGES_REQUESTED** [review]

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/attributes.adoc (hunk 4 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/attributes.adoc:14]: yes, typo!

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc (hunk 1 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:4]: done

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc (hunk 7 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:6]: done

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc (hunk 5 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:15]: @benradey will this be true in Pantheon v2?

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc (hunk 6 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:20]: done

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc (hunk 1 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc]: changed to "As a Pantheon user, you are assigned a role that determines what actions you can perform."

**Comment by theashiot** [comment]
Hi @stoobie,

I've incorporated your suggestions, have a look?

many thanks,

ashwin

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc (hunk 5 above)
Comment (benradey) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:15]: @stoobie @theashiot Managing products and product versions will be limited to publishers only in Pantheon 2.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc (hunk 6 above)
Comment (benradey) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc]: @stoobie If what you say is true, then you have likely found a bug. It's not intended that authors can manage products in Pv2.

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 8]
```diff
@@ -0,0 +1,20 @@
+[id="user-roles-in-pantheon_{context}"]
```
Comment (adahms) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:1]: Remove '-in-pantheon'

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+= User roles in {ProductShortName}
```
Comment (adahms) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:2]: Remove 'in {ProductShortName}'

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+As a {ProductShortName} user, you are assigned a role that determines what actions you can perform.
```
Comment (adahms) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles-in-pantheon.adoc:4]: Add the tag [role="_abstract"] directly above this line, no spaces in between.

**Review by adahms — CHANGES_REQUESTED** [review]
Hi @theashiot - added three easy comments to bring this in line with the changes we discussed as a team and to add the abstract tag.

**Comment by theashiot** [comment]
Hi, @adahms and @stoobie, I've made the suggested updates. Can you please review the updates?  Do you think we should keep doc as-is, wrt only Publishers being able to add product version? 

many thanks,

ashwin

**Review by adahms — DISMISSED** [review]
LGTM; approving.

Thanks @theashiot!

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles.adoc [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+* Revise existing content.
+* Preview and stage content.
+* Search and sort any content imported into {ProductShortName}.
+* Manage module metadata that is configurable in the user interface.
+ 
+*Publisher*::
+* All the permissions associated with the *Author* role.
+* Publish content to the {PublishingPortal} and the {PublishingPortalSearch}. 
+* Unpublish content from the {PublishingPortal} and the {PublishingPortalSearch}.
+* Create and manage product and product version.
```
Comment (stoobie) [inline pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles.adoc:21]: `* Create and manage {ProductShortName} product and product version.`

**Review by stoobie — CHANGES_REQUESTED** [review]

File: pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles.adoc (hunk 11 above)
Comment (theashiot) [reply pantheon-bundle/src/main/resources/SLING-INF/content/docs/modules/con_user-roles.adoc]: done!

**Comment by stoobie** [comment]
LGTM

**Review by stoobie — APPROVED** [review]
LGTM

**Comment by theashiot** [comment]
Thanks @stoobie!

@redhataccess/pantheon-developers. Please review and merge.

best,

ashwin

**Comment by adahms** [comment]
Hey @benradey - can someone from the engineering team merge this one for us?

**Review by benradey — APPROVED** [review]

**Pull request merged**